### PR TITLE
Run3-alca249 Modify calibration macros to use the current IsoTrack calibration

### PR DESCRIPTION
**Pull request by bsunanda**
#### PR description:

Modify calibration macros to use the current IsoTrack calibration. We introduce a new mode (truncate flag 9) in place of its earlier implementation. We have a new depth-dependent IsoTrack calibration where parameters for depth = 1 and 2 are kept the same 

#### PR validation:

Use the macros for the current IsoTrack analysis

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

Nothing special

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@cmsbuild Please test

File: Calibration/HcalCalibAlgos/macros/CalibCorr.C [hunk 1]
```diff
@@ -189,13 +189,9 @@ unsigned int truncateId(unsigned int detId, int truncateFlag, bool debug = false
...
     }
   } else if (truncate0 == 9) {
-    //Depths 1 and 2 in HB considered as depth = 1; rest of the depths in HB as depth = 2
-    if (subdet == 1) {
-      if (depth > 1)
-        depth = 2;
-      else
-        depth = 1;
-    }
+    //Depths 1 and 2 considered as depth = 1; rest of the depths are kept as it is
```
Comment (perrotta) [inline Calibration/HcalCalibAlgos/macros/CalibCorr.C:192]: The comment does not correspond to what is implemented in the code (it should be 
```suggestion
    //Depths 1 and 2 considered as depth = 2; rest of the depths are kept as it is
```
instead)

And in any case the implementation here is quite different from what was before: please check and explain what was updated in HB and HE that requires this change

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by perrotta** [comment]
@bsunanda, just for the record this PR changes the implementation as follows:

=== HB moved from
depth_current depth_withPR
     1            1
     2            2
     3            2
     4            2
to
depth_current depth_withPR
     1            2
     2            2
     3            3
     4            4

=== HE moved from
```
depth_current depth_withPR
     1            1
     2            2
     3            3
     4            4
to
depth_current depth_withPR
     1            2
     2            2
     3            3
     4            4

**Comment by perrotta** [comment]
@bsunanda just to be clear, this PR changes the implementation as follows

=== **HB** moved from
```
depth_current depth_withPR
     1            1
     2            2
     3            2
     4            2
```
to
```
depth_current depth_withPR
     1            2
     2            2
     3            3
     4            4
```

=== **HE** moved from
```
depth_current depth_withPR
     1            1
     2            2
     3            3
     4            4
```
to
```
depth_current depth_withPR
     1            2
     2            2
     3            3
     4            4
```

I assume that this is what you want, but since it is not explained anywhere in the PR description please confirm (even better if you can link where in the "current IsoTrack calibration" this is implemented).

**Comment by perrotta** [comment]
+alca

**Comment by mandrenguyen** [comment]
+1

**Pull request merged**